thewalrus/internal_modes/useful_tools.py

```diff
@@ -0,0 +1,140 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Useful tools for dealing with internal modes"""
+import numpy as np
+from scipy.special import factorial
+import numba
+from ..charpoly import powertrace
+
+fact = np.array([factorial(i) for i in range(110)], dtype=np.float64)
```

Comment:
Why find the factorials up to 110? Wherever this is called, could we not choose the upper limit based on the detection pattern and cutoff (ie only calculate what is needed)?